spyder/plugins/profiler/widgets/main_widget.py

```diff
@@ -31,156 +31,339 @@
 from spyder.widgets.comboboxes import PythonModulesComboBox
-from spyder.utils.misc import add_pathlist_to_PYTHONPATH, getcwd_or_home
-from spyder.plugins.variableexplorer.widgets.texteditor import TextEditor
-
-# This is needed for testing this module as a stand alone script
-try:
-    _ = get_translation("profiler", "spyder_profiler")
-except KeyError as error:
-    import gettext
-    _ = gettext.gettext
 
+# Localization
+_ = get_translation('spyder')
 
+# Logging
 logger = logging.getLogger(__name__)
+
+
+# --- Constants
+# ----------------------------------------------------------------------------
+if is_dark_interface():
+    MAIN_TEXT_COLOR = 'white'
+else:
+    MAIN_TEXT_COLOR = '#444444'
+
+
+class ProfilerWidgetActions:
+    # Triggers
+    Browse = 'browse_action'
+    Clear = 'clear_action'
+    Collapse = 'collapse_action'
+    Expand = 'expand_action'
+    LoadData = 'load_data_action'
+    Run = 'run_action'
+    SaveData = 'save_data_action'
+    ShowOutput = 'show_output_action'
+
+
+class ProfilerWidgetToolBars:
+    Information = 'information_toolbar'
+
+
+class ProfilerWidgetMainToolBarSections:
+    Main = 'main_section'
+
+
+class ProfilerWidgetInformationToolBarSections:
+    Main = 'main_section'
+
+
+# --- Utils
+# ----------------------------------------------------------------------------
 def is_profiler_installed():
     from spyder.utils.programs import is_module_installed
     return is_module_installed('cProfile') and is_module_installed('pstats')
 
 
-class ProfilerWidget(QWidget):
+def gettime_s(text):
     """
-    Profiler widget
+    Parse text and return a time in seconds.
+
+    The text is of the format 0h : 0.min:0.0s:0 ms:0us:0 ns.
+    Spaces are not taken into account and any of the specifiers can be ignored.
     """
-    DATAPATH = get_conf_path('profiler.results')
-    VERSION = '0.0.1'
-    redirect_stdio = Signal(bool)
+    pattern = r'([+-]?\d+\.?\d*) ?([munsecinh]+)'
+    matches = re.findall(pattern, text)
+    if len(matches) == 0:
+        return None
+    time = 0.
+    for res in matches:
+        tmp = float(res[0])
+        if res[1] == 'ns':
+            tmp *= 1e-9
+        elif res[1] == 'us':
+            tmp *= 1e-6
+        elif res[1] == 'ms':
+            tmp *= 1e-3
+        elif res[1] == 'min':
+            tmp *= 60
+        elif res[1] == 'h':
+            tmp *= 3600
+        time += tmp
+    return time
 
-    def __init__(self, parent, max_entries=100, options_button=None,
-                 text_color=None):
-        QWidget.__init__(self, parent)
 
-        self.setWindowTitle("Profiler")
+# --- Widgets
+# ----------------------------------------------------------------------------
+class ProfilerWidget(PluginMainWidget):
+    """
+    Profiler widget.
+    """
+    DEFAULT_OPTIONS = {
+        'text_color': MAIN_TEXT_COLOR,
+    }
+    DATAPATH = get_conf_path('profiler.results')  # Move this to be an option

```

Comment:
># Move this to be an option

The idea is that there should be a better way to expose the use of `get_conf_path`, so that is what I mean. Importing that from the main widget is weird.

Reply:
Will remove for now and address globally (for other plugins) on another PR